Content.Server/Paper/PaperSystem.cs

```diff
@@ -147,6 +150,14 @@ private void OnInputTextMessage(EntityUid uid, PaperComponent paperComp, PaperIn
             {
                 paperComp.Content = args.Text;
 
+                var ev = new WriteAttemptEvent(args.Session.AttachedEntity, uid);
+                RaiseLocalEvent<WriteAttemptEvent>(uid, ev);
```

Comment:
1. Don't need the generic.
2. These events should be record structs raised by-ref.

Reply:
Removed the events, callers use the method directly now (see comment on IlliterateSystem)